frontend/packages/dev-console/src/components/user-preferences/SecureRouteFields.tsx

```diff
@@ -0,0 +1,184 @@
+        secure,
+        tlsTermination,
+        insecureTraffic: selection,
+      });
+      setIsInsecureTrafficOpen(false);
+    },
+    [secure, setPreferredRoutingOptions, tlsTermination],
+  );
+
+  return preferredRoutingOptionsLoaded ? (
+    <div className="pf-c-form">
+      <span className="co-help-text">
+        {t(
+          'devconsole~The defaults below will only apply to the Import from Git and Deploy Image forms when creating Deployments or Deployment Configs.',
+        )}
+      </span>
+      <Checkbox
+        id="secure-route-checkbox"
+        data-test="secure-route-checkbox"
+        label={t('devconsole~Secure route')}
+        isChecked={secure}
+        data-checked-state={secure}
+        onChange={onSecureRouteChecked}
+        aria-label={t('devconsole~Secure route')}
+        description={t(
+          'devconsole~Routes can be secured using several TLS termination types for serving certificates.',
+        )}
+        className="odc-secure-route-fields__secure-route"
+      />
+
+      <FormGroup fieldId="tls-termination" label={t('devconsole~TLS termination')}>
+        <Select
+          id="tls-termination"
+          variant={SelectVariant.single}
+          isOpen={isTLSTerminationOpen}
+          selections={tlsTermination}
+          toggleId="tls-termination"
+          onToggle={onTLSTerminationToggle}
+          onSelect={onTLSTerminationSelect}
+          placeholderText={t('devconsole~Select termination type')}
+          aria-label={t('devconsole~Select termination type')}
+          maxHeight={300}
+        >
+          {tlsTerminationSelectOptions}
+        </Select>
+      </FormGroup>
+      <FormGroup
+        fieldId="insecure-traffic"
+        label={t('devconsole~Insecure traffic')}
+        helperText={t('devconsole~Policy for traffic on insecure schemes like HTTP.')}
+      >
+        <Select
+          id="insecure-traffic"
+          variant={SelectVariant.single}
+          isOpen={isInsecureTrafficOpen}
+          selections={insecureTraffic}
+          toggleId="insecure-traffic"
+          onToggle={onInsecureTrafficToggle}
+          onSelect={onInsecureTrafficSelect}
+          placeholderText={t('devconsole~Select insecure traffic type')}
+          aria-label={t('devconsole~Select insecure traffic type')}
+          maxHeight={300}
+        >
+          {tlsTermination === 'passthrough'
+            ? passthroughInsecureTrafficTypesSelectOptions
+            : insecureTrafficSelectOptions}
+        </Select>
+      </FormGroup>
+    </div>
+  ) : (
+    <>
+      <Skeleton
+        height="15px"
+        width="100%"
+        id="skeleton-secureRoute"
+        className="odc-secure-route-fields__secure-route"
+      />
+      <Skeleton
+        height="30px"
+        width="100%"
+        id="skeleton-tlsTermination"
+        className="odc-secure-route-fields__secure-route"
+      />
+      <Skeleton height="30px" width="100%" id="skeleton-insecureTraffic" />
+    </>
```

Comment:
This code looks not bad, but it doesn't result in the expected layout. Aritra has added Skeletons so that the other forms look like this if loaded is false. (I set this locally to test this and make a screenshot.)

![image](https://user-images.githubusercontent.com/139310/146565195-b1836d1d-e226-4e41-9609-0754a9e69513.png)

But these skeletons here are rendered like this:

![image](https://user-images.githubusercontent.com/139310/146565300-0efb6e09-6b85-4043-9dbf-1e9556e0839e.png)

Nothing to see. So we can just return null, or should implement this in a way these forms are rendered correctly.

I see three options:

1. Just render nothing:
```tsx
if (!preferredRoutingOptionsLoaded) {
  return null;
}
```

2. Render the `Skeleton`s similar to the other user preferences fields. `UserPreferenceDropdownField` renders only for the input element, the `FormGroup` is part of a parent component, so you need to keep it as well.

```tsx
  return (
    <div className="pf-c-form">
      <span className="co-help-text">
        {t(....)}
      </span>
      {preferredRoutingOptionsLoaded ? (
        <Checkbox ... />
      ) : (
        <Skeleton />
      )}
      <FormGroup>
        {preferredRoutingOptionsLoaded ? (
          <Select ... />
        ) : (
          <Skeleton />
        )}
      </FormGroup>
      <FormGroup>
        {preferredRoutingOptionsLoaded ? (
          <Select ... />
        ) : (
          <Skeleton />
        )}
      </FormGroup>
    </div>
  );
```

3. Disable the fields until the content is loaded. This reduces the JSX code, is straightforward, but looks a little bit different from the other user settings. But the loading indicator is a detail that is normally not visible for this setting.

```tsx
  return (
    <div className="pf-c-form">
      <span className="co-help-text">
        {t(....)}
      </span>
      <Checkbox ... isDisabled={!preferredRoutingOptionsLoaded} />
      <FormGroup>
         <Select ... isDisabled={!preferredRoutingOptionsLoaded} />
      </FormGroup>
      <FormGroup>
        <Select ... isDisabled={!preferredRoutingOptionsLoaded} />
      </FormGroup>
    </div>
  );
```

Reply:
Updated the PR with 3rd option.